internal/driver/glfw/window.go

```diff
@@ -621,11 +622,19 @@ func (w *window) mouseClicked(_ *glfw.Window, btn glfw.MouseButton, action glfw.
 		}
 	}
 
+	needsfocus := false
 	if layer != 1 { // 0 - overlay, 1 - menu, 2 - content
-		if wid, ok := co.(fyne.Focusable); ok {
-			w.canvas.Focus(wid)
-		} else {
-			w.canvas.Unfocus()
+		needsfocus = true
+
+		if wid := w.canvas.Focused(); wid != nil {
+			if wid.(fyne.CanvasObject) != co {
+				// Avoid changing focus when a toolbar button is pressed.
+				if _, isToolbarButton := co.(*widget.ToolbarButton); !isToolbarButton {
```

Comment:
This is confusing - if you made a new ToolbButton so that it didn't accept focus like normal buttons then this should not be needed

Reply:
Unfortunatly it cannot. The toolbar must be focusable by TAB key-presses, but should not be focused when tapped. On the other hand, the code can be simplified as I have now done.